src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/ReflectionEmitCachingMemberAccessor.Cache.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if NETFRAMEWORK || NETCOREAPP
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.Text.Json.Serialization.Metadata
+{
+    internal sealed partial class ReflectionEmitCachingMemberAccessor
+    {
+        private sealed class Cache<TKey> where TKey : notnull
+        {
+            private int _invocationCount; // tracks total number of invocations to the cache; can be allowed to overflow.
```

Comment:
@stephentoub per offline discussion I did a couple of things here:

- Used an unbounded cache evicting entries based on sliding expiration. I ran a few benchmarks using a bounded MRU cache and which showed receding performance gains as the number of members grew relative to the fixed maximum cache size.
- Instead of using `Timer` to schedule evictions at a fixed interval, schedules evictions synchronously every _n_ th read attempt.

The obvious drawback of this approach is that eviction is iterating through all the dictionary context, which in certain access patterns can be expensive.